IWYU errors in CI

IWYU in CI is reporting that there are unused header files, such as in https://github.com/librsync/librsync/actions/runs/14822623417/job/41611649020#step:6:11

```
cmake --build . --target iwyu
  shell: /usr/bin/bash -e {0}
[100%] Check #includes for correctness using iwyu_tool.
/home/runner/work/librsync/librsync/tests/sumset_test.c:28:1: error: superfluous '#include "hashtable.h"'
gmake[[3](https://github.com/librsync/librsync/actions/runs/14822488789/job/41611359535#step:6:3)]: *** [CMakeFiles/iwyu.dir/build.make:71: CMakeFiles/iwyu] Error 1
gmake[2]: *** [CMakeFiles/Makefile2:216: CMakeFiles/iwyu.dir/all] Error 2
gmake[1]: *** [CMakeFiles/Makefile2:223: CMakeFiles/iwyu.dir/rule] Error 2
gmake: *** [Makefile:218: iwyu] Error 2
Error: Process completed with exit code 2.
```

I can't reproduce this locally with iwyu 8.17-1 from Debian: in fact if I remove that line then it complains that it ought to be added. CI seems to have 8.21-1build2.

The test needs the `hashtable_t` type, which is defined in `hashtable.h` but that file is indirectly included by `sumset.h`. Maybe it's getting confused or has changed its behavior with transiently included declarations.

As I understand it, the idea IWYU and the standard at Google is to avoid transient headers and to include only what's needed, which makes sense in a huge monorepo. I'm not so sure it matters here. 

@dbaarda I'm inclined to drop it from CI, while still letting it be run manually...

--- a/.github/workflows/lint.yml
+++ b/.github/workflows/lint.yml
@@ -21,5 +21,7 @@ jobs:
     - name: clang-tidy
       run: cmake --build . --target clang-tidy
 
-    - name: iwyu
-      run: cmake --build . --target iwyu
+    # We don't run IWYU from CI because it's oriented to codebases with no recursive includes,
+    # but librsync does have recursive includes.
+    # - name: iwyu
+    #   run: cmake --build . --target iwyu

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -261,6 +261,10 @@ add_custom_target(clang-tidy
 
 # iwyu target to check includes for correctness.
 # Note we ignore noisy "note:" output.
+#
+# Note that iwyu is oriented to codebases with no recursive includes,
+# but librsync does currently have recursive includes; this may give warnings
+# for some files that are not practical problems.
 add_custom_target(iwyu
     COMMENT "Check #includes for correctness using iwyu_tool."
     COMMAND ! iwyu_tool -p ${CMAKE_CURRENT_BINARY_DIR} -o clang | egrep -v "note:"

